catroid/src/main/java/org/catrobat/catroid/scratchconverter/protocol/message/job/JobMessage.java

```diff
@@ -0,0 +1,145 @@
+
+	private static final String TAG = JobMessage.class.getSimpleName();
+
+	private final long jobID;
+
+	public enum Type {
+		JOB_FAILED(0),
+		JOB_RUNNING(1),
+		JOB_ALREADY_RUNNING(2),
+		JOB_READY(3),
+		JOB_OUTPUT(4),
+		JOB_PROGRESS(5),
+		JOB_FINISHED(6);
+
+		private int typeID;
+
+		private static Map<Integer, Type> map = new HashMap<>();
+		static {
+			for (Type legEnum : Type.values()) {
+				map.put(legEnum.typeID, legEnum);
+			}
+		}
+		Type(final int typeID) {
+			this.typeID = typeID;
+		}
+
+		public static Type valueOf(int typeID) {
+			return map.get(typeID);
+		}
+
+		public int getTypeID() {
+			return typeID;
+		}
+	}
+
+	public JobMessage(final long jobID) {
+		this.jobID = jobID;
+	}
+
+	public long getJobID() {
+		return jobID;
+	}
+
+	@Nullable
+	public static <T extends JobMessage> T fromJson(JSONObject jsonMessage) throws JSONException {
+		final JSONObject jsonData = jsonMessage.getJSONObject(JsonKeys.DATA.toString());
+		final long jobID = jsonData.getLong(JsonDataKeys.JOB_ID.toString());
+		switch (Type.valueOf(jsonMessage.getInt(JsonKeys.TYPE.toString()))) {
+			case JOB_FAILED:
+				return (T) new JobFailedMessage(jobID, jsonData.getString(JsonDataKeys.MSG.toString()));
+
+			case JOB_RUNNING:
+				final String jobTitle = jsonData.getString(JsonDataKeys.JOB_TITLE.toString());
+				final String jobImageURLKey = JsonDataKeys.JOB_IMAGE_URL.toString();
+				final String jobImageURL = jsonData.isNull(jobImageURLKey) ? null : jsonData.getString(jobImageURLKey);
+				return (T) new JobRunningMessage(jobID, jobTitle, jobImageURL);
+
+			case JOB_ALREADY_RUNNING:
+				final String jTitle = jsonData.getString(JsonDataKeys.JOB_TITLE.toString());
+				final String jImageURLKey = JsonDataKeys.JOB_IMAGE_URL.toString();
+				final String jImageURL = jsonData.isNull(jImageURLKey) ? null : jsonData.getString(jImageURLKey);
+				return (T) new JobAlreadyRunningMessage(jobID, jTitle, jImageURL);
+
+			case JOB_READY:
+				return (T) new JobReadyMessage(jobID);
+
+			case JOB_OUTPUT:
+				final JSONArray jsonLines = jsonData.getJSONArray(JsonDataKeys.LINES.toString());
+				final List<String> lineList = new ArrayList<>();
+				for (int i = 0; i < jsonLines.length(); ++i) {
+					lineList.add(jsonLines.getString(i));
+				}
+				final String[] lines = lineList.toArray(new String[lineList.size()]);
+				return (T) new JobOutputMessage(jobID, lines);
+
+			case JOB_PROGRESS:
+				final short progress = (short) jsonData.getInt(JsonDataKeys.PROGRESS.toString());
+				return (T) new JobProgressMessage(jobID, progress);
+
+			case JOB_FINISHED:
+				final String downloadURL = jsonData.getString(JsonDataKeys.URL.toString());
+				final String cachedUTCDateKey = JsonDataKeys.CACHED_UTC_DATE.toString();
+				final String dateUTC = jsonData.isNull(cachedUTCDateKey) ? null : jsonData.getString(cachedUTCDateKey);
+				final DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.US);
```

Comment:
Please create a constant for the DateFormat   (DATE_FORMAT_ISO_8601 doesn't match exactly)